pkg/block/fetcher.go

```diff
@@ -107,7 +109,7 @@ type GaugeLabeled interface {
 	WithLabelValues(lvs ...string) prometheus.Gauge
 }
 
-type MetaFetcherFilter func(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, incompleteView bool)
+type MetaFetcherFilter func(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, incompleteView bool) error
```

Comment:
@bwplotka Would it make sense to have this as an interface, instead of just type? Perhaps it's just a tooling problem, it took me a while to realize that various `Filter` methods have this type. With interface, the connection is clearer and my IDE of choice (Goland) would actually show this connection between implementation and interface directly.

Reply:
Agree with this, just that this won't be part of this PR.

Reply:
I don't get why interface helps here until we got into casting or multiple methods. I would keep it function as it's much more simpler to just pass function. But I agree we have just structs anywere... Not strong opinon.

Reply:
I am resolving this thread since this will not be done as part of this PR.